R/create_credentials.R

```diff
@@ -120,6 +126,39 @@ create_smtp_creds_key <- function(id,
   # Determine whether the keyring package can be used
   validate_keyring_capable()
 
+  # Stop function if `id` provided is not of the right type
+  if (!(inherits(id, "character") | inherits(id, "numeric") | inherits(id, "integer"))) {
+    stop("The provided `id` value must be a single character or numeric value.",
+         call. = TRUE)
+  }
+
+  # Stop function if the `id` value isn't of length 1
+  if (length(id) != 1) {
+    stop("Only a vector of length 1 should be used for the `id` value.",
+         call. = FALSE)
+  }
+
+  # Stop function if the `id` is an empty string or NA
+  if (id == "" || is.na(id)) {
```

Comment:
`id == "" || is.na(id)` should be `is.na(id) || id == ""`

Reply:
Made this change. Thanks.